### PR TITLE
chore: cleaning

### PR DESCRIPTION
**Pull request by drdelambre**
working on cleaning up some of the prop drilling going on in here

also turns typescript back on

### CODE REVIEW

File: ui/src/pageLayout/components/CloudNav.tsx [hunk 1]
```diff
@@ -37,6 +36,8 @@ interface StateProps {
   org: Organization
 }
 
+const Logo = require('../images/influxdata-logo.png')
```
Comment (drdelambre) [inline ui/src/pageLayout/components/CloudNav.tsx]: idk how the import worked earlier, but this works now

File: ui/tsconfig.json [hunk 2]
```diff
@@ -35,11 +35,15 @@
     "coverage",
     "jest.config.js",
     "webpack.tsconfig.json",
-    "webpack.star.ts",
+    "webpack.*.ts",
```
Comment (drdelambre) [inline ui/tsconfig.json]: no idea

File: ui/tsconfig.json [hunk 3]
```diff
@@ -35,11 +35,15 @@
...
     "src/**/*.test.ts",
     "src/**/*.test.tsx",
     "src/**/mocks.ts",
     "coverage"
   ],
-  "include": ["global.d.ts"]
+  "include": [
```
Comment (drdelambre) [inline ui/tsconfig.json]: this turned typescript back on

File: ui/src/dashboards/components/DashboardHeader.tsx [hunk 4]
```diff
@@ -15,44 +17,73 @@ import {
...
-  handleChooseTimeRange: (timeRange: TimeRange) => void
-  handleChooseAutoRefresh: (autoRefreshInterval: number) => void
-  onSetAutoRefreshStatus: (status: AutoRefreshStatus) => void
   onManualRefresh: () => void
-  handleClickPresentationButton: AppActions.DelayEnablePresentationModeDispatcher
-  onAddCell: () => void
-  onAddNote: () => void
-  onRenameDashboard: (name: string) => void
-  toggleVariablesControlBar: () => void
-  isShowingVariablesControlBar: boolean
```
Comment (121watts) [inline ui/src/dashboards/components/DashboardHeader.tsx]: Yaaaaaaaaaaaaaaaaaas

File: ui/src/dashboards/components/Dashboard.tsx [hunk 5]
```diff
@@ -27,39 +26,23 @@ interface StateProps {
...
 @ErrorHandling
 class DashboardComponent extends PureComponent<Props> {
   public render() {
-    const {
-      cells,
-      status,
-      timeRange,
-      manualRefresh,
-      onPositionChange,
-      onAddCell,
```
Comment (121watts) [inline ui/src/dashboards/components/Dashboard.tsx]: YAAAAAAAAAAAAAAAAAAAS

File: ui/src/dashboards/components/DashboardPage.tsx [hunk 6]
```diff
@@ -14,255 +13,70 @@ import VariablesControlBar from 'src/dashboards/components/variablesControlBar/V
...
-  getDashboard: typeof dashboardActions.getDashboard
-  updateDashboard: typeof dashboardActions.updateDashboard
-  updateCells: typeof cellActions.updateCells
-  updateQueryParams: typeof rangesActions.updateQueryParams
-  setDashboardTimeRange: typeof rangesActions.setDashboardTimeRange
-  handleChooseAutoRefresh: typeof setAutoRefreshInterval
-  onSetAutoRefreshStatus: typeof setAutoRefreshStatus
-  handleClickPresentationButton: AppActions.DelayEnablePresentationModeDispatcher
-  onUpdateView: typeof updateViewAndVariables
-  onToggleShowVariablesControls: typeof toggleShowVariablesControls
```
Comment (121watts) [inline ui/src/dashboards/components/DashboardPage.tsx]: YAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAS

File: ui/src/dashboards/components/DashboardPage.tsx [hunk 7]
```diff
@@ -14,255 +13,70 @@ import VariablesControlBar from 'src/dashboards/components/variablesControlBar/V
...
-
-export default connect<StateProps, DispatchProps>(
+export default connect<StateProps, {}>(
   mstp,
-  mdtp
-)(
-  ManualRefresh<OwnProps>(
-    withRouter<OwnProps & ManualRefreshProps>(DashboardPage)
-  )
-)
```
Comment (121watts) [inline ui/src/dashboards/components/DashboardPage.tsx]: All of these deletions make me so happy

File: ui/src/dashboards/components/dashboard_empty/DashboardEmpty.tsx [hunk 8]
```diff
@@ -30,13 +25,21 @@ class DashboardEmpty extends Component<Props> {
...
+            onClick={this.handleAdd}
             testID="add-cell--button"
           />
         </EmptyState>
       </div>
     )
   }
+
+  private handleAdd = () => {
+    // TODO(alex): change this to using state from redux
```
Comment (121watts) [inline ui/src/dashboards/components/dashboard_empty/DashboardEmpty.tsx]: Let's do this now!  You should already have the `dashboardID` and `orgID` from the `WithRouterProps`

File: ui/src/shared/actions/currentDashboard.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+export type Action = ReturnType<typeof setDashboard>
+
+export const SET_DASHBOARD = 'SET_DASHBOARD_ID'
```
Comment (121watts) [inline ui/src/shared/actions/currentDashboard.ts]:
```suggestion
export const SET_CURRENT_DASHBOARD = 'SET_CURRENT_DASHBOARD'
```

File: ui/src/shared/components/DashboardRoute.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    if (dashboard !== dashboardID) {
+      updateDashboard(dashboardID)
+    }
+  }
+
+  componentDidMount() {
+    this.check(this.props)
+  }
+
+  componentWillUnmount() {
```
Comment (121watts) [inline ui/src/shared/components/DashboardRoute.tsx]: You could do all of this with the `useEffect` hook in an `FC`

File: ui/src/shared/reducers/currentDashboard.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import {Action, SET_DASHBOARD} from 'src/shared/actions/currentDashboard'
+
+export interface CurrentDashboardState {
+  dashboard: string
```
Comment (121watts) [inline ui/src/shared/reducers/currentDashboard.ts]:
```suggestion
id: string
```
`state.currentDashboard.dashboard` reads a bit oddly.  Also, I don't know if its the `id`, `name`, or the whole `dashboard`.  Also `state.currentDashboard.id` looks and sounds exactly like what it is.

File: ui/tsconfig.json (hunk 2 above)
Comment (121watts) [reply ui/tsconfig.json]: whaaaaaaaaaaaa

File: ui/src/shared/components/DashboardRoute.tsx (hunk 10 above)
Comment (drdelambre) [reply ui/src/shared/components/DashboardRoute.tsx]: i still don't know if this is the right way to do this.. so i'm trying to stay flexible. i think i want it as a decorator declared on the route, but.. here we are

File: ui/src/shared/reducers/currentDashboard.ts (hunk 11 above)
Comment (drdelambre) [reply ui/src/shared/reducers/currentDashboard.ts]: changed it to `id`. totally makes more sense

**Pull request merged**